src/main/java/io/cdap/plugin/servicenow/apiclient/ServiceNowTableAPIClientImpl.java

```diff
@@ -134,15 +134,13 @@ public List<Map<String, String>> fetchTableRecords(String tableName, SourceValue
         if (apiResponse.isRetryable()) {
           throw new RetryableException();
         }
-        return Collections.emptyList();
       }
 
       return parseResponseToResultListOfMap(apiResponse.getResponseBody());
     } catch (OAuthSystemException e) {
       throw new RetryableException();
     } catch (OAuthProblemException e) {
```

Comment:
I believe the system vs problem would be around system being busy vs problem being credential being wrong.

Reply:
`OAuthSystemException` is when a system problem occurs, so we retry on this exception. `OAuthProblemException` is when authentication fails.